tools/refguide_check.py

```diff
@@ -645,6 +645,10 @@ def write(self, msg):
             else:
                 out(msg)
 
+        def flush(self):
```

Comment:
why is this required?

Reply:
I added this in response to a CI failure. However, since rebasing on master I don't know how to retrieve that failure to demonstrate and I can't remember more details than that.
I'll commit the other changes that you mentioned, then try and remove this.

Reply:
There was a reason I added it see this [travisCI line](https://travis-ci.org/andyfaff/scipy/jobs/423775764#L1378).

Reply:
I'll do a massive squish of all the commits when it's decided the PR is good to go.

Reply:
interesting failure. could you add a comment here so we're not puzzled again by this later?

Reply:
Done.